test/syncenginetestutils.h

```diff
@@ -706,17 +706,26 @@ class FakeErrorReply : public QNetworkReply
 
 class FakeQNAM : public QNetworkAccessManager
 {
+public:
+    using Observer = std::function<void(QByteArray verb, QString fileName)>;
```

Comment:
And in fact, it could return a QNetworkReply *,  which would be returned if it is not nullptr

So some tests could throw custom error